graph-service/curator.py

```
@@ -0,0 +1,565 @@
+                json_str = output.strip()
+                if json_str.startswith("```"):
+                    lines = json_str.split("\n")
+                    json_str = "\n".join(lines[1:-1]) if len(lines) > 2 else json_str
+                elif json_str.startswith("```json"):
+                    lines = json_str.split("\n")
+                    json_str = "\n".join(lines[1:-1]) if len(lines) > 2 else json_str
+                
+                data = json.loads(json_str)
+                return CuratorAnalysis(**data)
+            except (json.JSONDecodeError, ValidationError) as e:
+                logger.error(f"Failed to parse curator analysis: {e}\nResponse: {output[:500]}")
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)